blocks/extensions/__init__.py

```diff
@@ -92,18 +107,22 @@ class SimpleExtension(TrainingExtension):
         If ``True``, :meth:`do` is invoked before the first epoch.
     after_every_epoch : bool
         If ``True``, :meth:`do` is invoked after every epoch.
-    after_every_iteration : bool
-        If ``True``, :meth:`do` is invoked after every iteration.
+    after_every_batch: bool
```

Comment:
It seems like this argument list could grow very long... Is there maybe an easier way e.g. just ask the user to pass the string directly for all cases but the `after_n_epochs`?